packages/api/src/submittable/createClass.ts

```diff
@@ -325,6 +325,7 @@ export function createClass <ApiType extends ApiTypes> ({ api, apiType, blockHas
 
     #signViaSigner = async (address: Address | string | Uint8Array, options: SignatureOptions, header: Header | null): Promise<SignerInfo> => {
       const signer = options.signer || api.signer;
+      const allowCallDataAlteration = options.allowCallDataAlteration || true;
```

Comment:
```suggestion
      const allowCallDataAlteration = options.allowCallDataAlteration || false;
```

Did you mean to do the above?

Reply:
Nah, I'd put this false, as in, per default, we now allow the call data to be changed.
If we flag this change as MAJOR, then we hope users to check the change log, and use this flag, set to `false` if they absolutely want to not allow external signers to change the call data.

Reply:
If that is the case then the following above will always be true, and something like:

```javascript
const allowCallDataAlteration = options.allowCallDataAlteration ?? true;
```

is more fitting.

Reply:
oh you are absolutely right. I'm too used to default params.
```suggestion
      const allowCallDataAlteration = options.allowCallDataAlteration ?? true;
```